libcontainer/rootfs_linux.go

```diff
@@ -557,11 +557,17 @@ func getCgroupMounts(m *configs.Mount) ([]*configs.Mount, error) {
 	return binds, nil
 }
 
-// checkProcMount checks to ensure that the mount destination is not over the top of /proc.
-// dest is required to be an abs path and have any symlinks resolved before calling this function.
+// Taken from <include/linux/proc_ns.h>. If a file is on a filesystem of type
+// PROC_SUPER_MAGIC, we're guaranteed that only the root of the superblock will
+// have this inode number.
+const procRootIno = 1
+
+// checkProcMount checks to ensure that the mount destination is not over the
+// top of /proc. dest is required to be an abs path and have any symlinks
+// resolved before calling this function.
 //
-// if source is nil, don't stat the filesystem.  This is used for restore of a checkpoint.
-func checkProcMount(rootfs, dest, source string) error {
+// source is "" when doing criu restores.
+func checkProcMount(rootfs, dest string, m *configs.Mount, source string) error {
```

Comment:
How about remove the param ‘source’, I think maybe it equals to ‘m.source’?

Reply:
It's not always `m.Source`. For bind-mounts, it's `/proc/self/fd/...` (1.1 doesn't have `mountEntry` and I didn't want to backport it because it would require too much extra work).

For criu-restore, the `""` value now has no effect, but I kept the logic because I wasn't comfortable removing it in a backport to 1.1.